src/telliot/utils/contract.py

```diff
@@ -23,17 +24,100 @@ class Contract(Base):
     #: ABI specifications of contract
     abi: List[Dict[str, Any]]
 
-    #: Contract address
-    web3_contract: Optional[web3.contract.Contract]
+    #: web3 contract object
+    contract: Optional[web3.contract.Contract]
 
-    class Config:
-        arbitrary_types_allowed = True
+    #: global pytelliot configurations
+    # config: ConfigOptions
 
-    def connect(self) -> None:
+    def connect(self) -> bool:
         """Connect to EVM contract through an RPC Endpoint"""
         if self.node.web3 is None:
-            raise TypeError("node is not connected")
+            print("node is not instantiated")
+            return False
         else:
-            self.web3_contract = self.node.web3.eth.contract(
+            if not self.node.connect():
+                print("node is not connected")
+                return False
+            self.contract = self.node.web3.eth.contract(
                 address=self.address, abi=self.abi
             )
+            return True
+
+    def read(self, func_name: str, **kwargs: Any) -> Tuple[Any, bool]:
+        """
+        Reads data from contract
+        inputs:
+        func_name (str): name of contract function to call
+
+        returns:
+        Tuple: contains contract function outputs (tuple) and success (bool)
+        """
+
+        if self.contract:
+            try:
+                contract_function = self.contract.get_function_by_name(func_name)
+                print(contract_function)
+                return (contract_function(**kwargs).transact(), True)
```

Comment:
remove redundant parentheses (on all tuple returns in this file)

Reply:
Thank you!